pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -187,14 +190,16 @@ func (m *ProvisioningClusterMutator) handlePSACT(request *admission.Request, clu
 				secretAnnotation: cluster.Name,
 			}
 			data := map[string][]byte{
-				secretKey: fileContent,
+				SecretKey: fileContent,
 			}
 			err = m.ensureSecret(cluster.Namespace, secretName, data, anno)
 			if err != nil {
 				return nil, fmt.Errorf("[provisioning cluster mutator] failed to create or update the secret for the admission configuration file: %w", err)
 			}
-			// set relevant fields in the cluster
-			addMachineSelectorFile(MachineSelectorFileForPSA(secretName, mountPath), cluster)
+			// drop then set relevant fields if they exist in the cluster
+			dropMachineSelectorFile(MachineSelectorFileForPSA(secretName, mountPath, ""), cluster, true)
+			hash := sha256.Sum256(fileContent)
+			addMachineSelectorFile(MachineSelectorFileForPSA(secretName, mountPath, base64.StdEncoding.EncodeToString(hash[:])), cluster)
```

Comment:
`hash[:]` this is pretty cool, I never thought about doing this to convert an array to a slice.